src/main/java/org/opensearch/geospatial/ip2geo/action/RestPutDatasourceHandler.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geospatial.ip2geo.action;
+
+import static org.opensearch.geospatial.shared.URLBuilder.URL_DELIMITER;
+import static org.opensearch.geospatial.shared.URLBuilder.getPluginURLPrefix;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.geospatial.ip2geo.common.Ip2GeoSettings;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+/**
+ * Rest handler for Ip2Geo datasource creation
+ *
+ * This handler handles a request of
+ * PUT /_plugins/geospatial/ip2geo/datasource/{id}
+ * {
+ *     "endpoint": {endpoint},
+ *     "update_interval_in_days": 3
```

Comment:
Can we make this of type "time" so that we dont have to include "in_days" in the parameter?

Reply:
Is there type "time"? Or do we want to create our own rule to parse it?

Reply:
https://github.com/opensearch-project/OpenSearch/blob/main/libs/common/src/main/java/org/opensearch/common/unit/TimeValue.java#L370